src/item.cpp

```diff
@@ -4831,6 +4831,24 @@ void item::melee_combat_info( std::vector<iteminfo> &info, const iteminfo_query
     }
 
     if( dmg_bash || dmg_cut || dmg_stab ) {
+        int stam;
+        float stam_pct;
+        std::map<std::string, double> dps_data;
+
+        bool base_dps = parts->test( iteminfo_parts::BASE_DPS );
```

Comment:
this is a bit nitpicky and personal taste but I think moving these to bools instead of just doing the test for the iteminfo_parts directly where you need it makes it harder to parse. Especially since all the other UI stuff does it directly.

Reply:
oh, i just didnt want to call it twice if it wasn't needed